skyportal/handlers/api/annotation.py

```diff
@@ -30,26 +51,65 @@ def get(self, annotation_id):
               application/json:
                 schema: Error
         """
-        annotation = Annotation.get_if_accessible_by(
-            annotation_id, self.current_user, raise_if_none=True
-        )
+
+        try:
+            annotation_id = int(annotation_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid (scalar integer) annotation ID. ")
+
+        if associated_resource_type.lower() == "sources":
+            try:
+                annotation = Annotation.get_if_accessible_by(
+                    annotation_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error('Could not find any accessible annotations.')
+            annotation_resource_id_str = str(annotation.obj_id)
+        # add more options using elif
+        else:
+            return self.error(
+                f'Unsupported input "{associated_resource_type}" given as "associated_resource_type" argument.'
+            )
+
+        if annotation_resource_id_str != resource_id:
```

Comment:
Under which circumstances can this occur?

Reply:
It can happen if the user tries to get annotation X from object Y but annotation X is actually associated with object Z.

Reply:
In that case, we should simply return a not found.  We cannot leak that the other object exists, or that there is a comment with that ID.

Reply:
I'm not sure if this is actually leaking anything. 
If the annotation ID is valid and accessible to the user, then the user should know the annotation exists. 
If it doesn't exist or is inaccessible, the handler returns `Could not find any accessible annotations.` a few lines before this. 

If the annotation is accessible but the object ID (or spectrum ID or whatever) doesn't match, it doesn't tell you that such an object exists, only that it does not match the ID of the object for that annotation (which the user is allowed to view). 
In that case the user should be made aware that he has a problem with the path parameter.

Reply:
OK, I'll resolve